runtime/hydradx/src/evm/precompiles/erc20_mapping.rs

```diff
@@ -70,17 +70,39 @@ pub struct SetCodeForErc20Precompile;
 impl RegisterAssetHook<AssetId> for SetCodeForErc20Precompile {
 	fn on_register_asset(asset_id: AssetId) {
 		pallet_evm::AccountCodes::<Runtime>::insert(HydraErc20Mapping::encode_evm_address(asset_id), &hex!["00"][..]);
+
+		let code = hex!["00"];
+		let size = code[..].len() as u64;
+		let hash = H256::from(sp_io::hashing::keccak_256(&code[..]));
+		let code_metadata = pallet_evm::CodeMetadata { size, hash };
+		pallet_evm::AccountCodesMetadata::<Runtime>::insert(
+			HydraErc20Mapping::encode_evm_address(asset_id),
+			code_metadata,
+		);
 	}
 }
 
-impl frame_support::traits::OnRuntimeUpgrade for SetCodeForErc20Precompile {
+pub struct SetCodeMetadataForErc20Precompile;
+impl frame_support::traits::OnRuntimeUpgrade for SetCodeMetadataForErc20Precompile {
```

Comment:
this impl should remain the same as the on_register_asset is called

Reply:
Refactored. I extracted the code and created a function for it.